host/greatfet/commands/greatfet_i2c.py

```diff
@@ -16,15 +16,14 @@
 
 
 def main():
+    from greatfet.utils import GreatFETArgumentParser
+
     # Set up a simple argument parser.
-    parser = argparse.ArgumentParser(description="Utility for I2C communication via GreatFET")
-    parser.add_argument('-s', dest='serial', metavar='<serialnumber>', type=str,
-                        help="Serial number of device, if multiple devices", default=None)
-    parser.add_argument('-v', dest='verbose', action='store_true', help="Write data from file")
+    parser = GreatFETArgumentParser(description="Utility for I2C communication via GreatFET")
     parser.add_argument('-z', '--scan', action='store_true', help="Scan all possible i2c addresses")
-    parser.add_argument('-a', '--address', nargs=1, type=str, help="Address to transmit data to over the I2C Bus") 
-    parser.add_argument('-t', '--transmit', metavar='data', default=[], help="Data to transmit over the I2C Bus")
-    parser.add_argument('-r', '--receive_length', default=1, help="Number of bytes expecting to receive from the I2C Bus")
+    parser.add_argument('transmit', nargs='*', type=ast.literal_eval, default=[], help="Data to transmit over the I2C Bus <byte byte byte..>")
```

Comment:
Does this drop `-t` and require using `--transmit`?

Reply:
It drops both -t and --transmit, ie. gf i2c -a 0x40 -r 2 -v 0x01 0x02 0x15

Reply:
Oh, of course, you told me about this before you did it.  My mistake.